app/services/structural_generator.rb

```diff
@@ -0,0 +1,46 @@
+# typed: true
+# frozen_string_literal: true
+
+# This generates a DROStructural for a work
+class StructuralGenerator
+  extend T::Sig
+
+  sig { params(work: Work).returns(T.any(Cocina::Models::DROStructural, Cocina::Models::RequestDROStructural)) }
+  def self.generate(work:)
+    new(work: work).generate
+  end
+
+  sig { params(work: Work).void }
+  def initialize(work:)
+    @work = work
+  end
+
+  attr_reader :work
+
+  sig { returns(T.any(Cocina::Models::DROStructural, Cocina::Models::RequestDROStructural)) }
+  def generate
+    klass = work.druid ? Cocina::Models::DROStructural : Cocina::Models::RequestDROStructural
+    klass.new(
+      contains: build_filesets,
+      isMemberOf: [work.collection.druid]
```

Comment:
OK, so *this* is the change that adds the collection membership. Cool.

Reply:
Yes